### PR TITLE
Fix the collection of statistics for dual integrators.

### PR DESCRIPTION
**Pull request by GiorgioGiangaspero**
Fix the collection of statistics (number of RHS evaluations, number of psuedo-steps and number of p-multigrid cycles) for dual integrators.

### CODE REVIEW

File: pyfr/integrators/dual/pseudo/multip.py [hunk 1]
```diff
@@ -301,5 +304,22 @@ def pseudo_advance(self, tcurr):
             if self.mg_convmon(self.pintg, i, self._minniters):
                 break
 
+        # Update the number of p-multigrid cycles
+        self.npmgcycles += i+1
```
Comment (FreddieWitherden) [inline pyfr/integrators/dual/pseudo/multip.py]: Spacing around i + 1.

File: pyfr/integrators/dual/pseudo/multip.py [hunk 2]
```diff
@@ -301,5 +304,22 @@ def pseudo_advance(self, tcurr):
...
+
         # Update the dual-time stepping banks
         self.finalise_mg_advance(self.pintg._idxcurr)
+
+    def collect_stats(self, stats):
+        # Collect the stats for each level
+        for l in self.levels:
+            # Total number of RHS evaluations
+            stats.set('solver-time-integrator', 'nfevals-p{}'.format(l),
```
Comment (FreddieWitherden) [inline pyfr/integrators/dual/pseudo/multip.py]: + str(l) rather than format

File: pyfr/integrators/dual/pseudo/pseudosteppers.py [hunk 3]
```diff
@@ -90,7 +88,7 @@ def _pseudo_stepper_has_lerrest(self):
 
     @property
     def _stepper_nfevals(self):
```
Comment (FreddieWitherden) [inline pyfr/integrators/dual/pseudo/pseudosteppers.py]: Should it still be called _stepper_nfevals?

Comment (GiorgioGiangaspero) [reply pyfr/integrators/dual/pseudo/pseudosteppers.py]: changed to _pseudo_stepper_nfevals.

**Comment by FreddieWitherden** [comment]
Looks good.  Cases we need to test:
 - regular dual time stepping
 - dual time stepping with polynomial multigrid
 - adaptive dual time stepping
 - adaptive dual time stepping with polynomial multigrid

Also, please squish into a single commit.

File: pyfr/integrators/dual/pseudo/multip.py [hunk 4]
```diff
@@ -297,9 +304,26 @@ def pseudo_advance(self, tcurr):
...
                 break
 
         # Update the dual-time stepping banks
         self.finalise_mg_advance(self.pintg._idxcurr)
+
+    def collect_stats(self, stats):
+        # Collect the stats for each level
+        for l in self.levels:
+            # Total number of RHS evaluations
+            stats.set('solver-time-integrator', 'nfevals-p' + str(l),
```
Comment (FreddieWitherden) [inline pyfr/integrators/dual/pseudo/multip.py:321]: Use new syntax `f'nfevals-p{l}'`.

Comment (GiorgioGiangaspero) [reply pyfr/integrators/dual/pseudo/multip.py]: ok

File: pyfr/integrators/dual/pseudo/multip.py [hunk 5]
```diff
@@ -297,9 +304,26 @@ def pseudo_advance(self, tcurr):
...
+    def collect_stats(self, stats):
+        # Collect the stats for each level
+        for l in self.levels:
+            # Total number of RHS evaluations
+            stats.set('solver-time-integrator', f'nfevals-p{l}',
+                      self.pintgs[l]._pseudo_stepper_nfevals)
+
+            # Total number of pseudo-steps
+            stats.set('solver-time-integrator', f'npseudosteps-p{l}',
+                       self.pintgs[l].npseudosteps)
```
Comment (FreddieWitherden) [inline pyfr/integrators/dual/pseudo/multip.py:326]: Indentation.

Comment (GiorgioGiangaspero) [reply pyfr/integrators/dual/pseudo/multip.py:326]: ok

**Comment by FreddieWitherden** [comment]
Are you okay to squish?

**Pull request merged**